crates/holochain_sqlite/src/db/p2p_metrics.rs

```diff
@@ -0,0 +1,82 @@
+use crate::{
+    prelude::{DatabaseError, DatabaseResult},
+    sql::sql_p2p_metrics,
+};
+use holochain_zome_types::prelude::*;
+use kitsune_p2p::event::{MetricKind, MetricQuery, MetricQueryAnswer};
+use kitsune_p2p::*;
+use rusqlite::*;
+use std::{
+    num::TryFromIntError,
+    sync::Arc,
+    time::{Duration, SystemTime, SystemTimeError},
+};
+
+fn time_to_ns(t: SystemTime) -> DatabaseResult<u64> {
+    t.duration_since(std::time::UNIX_EPOCH.into())
+        .map_err(|e| DatabaseError::Other(e.into()))?
+        .as_nanos()
+        .try_into()
+        .map_err(|e: TryFromIntError| DatabaseError::Other(e.into()))
+}
+
+fn time_from_ns(ns: u64) -> SystemTime {
+    std::time::UNIX_EPOCH
+        .checked_add(Duration::from_nanos(ns))
+        .expect("Time must be after 1970 and before the end of the universe")
+}
```

Comment:
sqlite stores integers as `i64`s. Not that we're going to overflow time-wise, but might be good to pedantically make the types match.

Also, we've got other code storing timestamps as milliseconds, and yet other code storing our timestamps as binary blobs, we should standardize on something...

Reply:
Yeah... @freesig what do you think? `i64` nanoseconds will overflow in the year 2270, but gives total precision. Maybe we can split the difference and go with microseconds for a cool 300,000 years?

Reply:
There are still processors out there that can't even return nanosecond precision times. I don't think we'd have any problems where micros weren't good enough.